// paxos proto file

syntax = "proto3";

option objc_class_prefix = "RTG";

package paxos;

// Interface exported by the server.
service Chatter {
  rpc SendChatMessage (ChatRequest) returns(ChatReply) {}
}

service Paxos {
  rpc Prepare(PrepareSend) returns (PromiseReply){}
}

message PrepareSend{
  int32 proposal =1;
}

message PromiseReply{
  int32 highest_proposal = 1;
  int32 highest_value = 2;
}

message ChatRequest {
  string mesg = 1;
  int32 seq_num = 2;
}

message ChatReply {
  string mesg = 1;
}
